### ISSUE TITLE
PublicKey does not roundtrip in fuzzing mode

### ISSUE THREAD

**Issue by apoelstra** [rust-bitcoin/rust-secp256k1#486]
Surprised this wasn't found much earlier.

This line https://github.com/rust-bitcoin/rust-secp256k1/blob/master/secp256k1-sys/src/lib.rs#L881 mangles public keys when they're being parsed in fuzzing mode. Then they serialize inconsistently which is breaking round-trip fuzztests in Liquid.

**Comment by jhfrontz** [comment]
This seems to be getting triggered much more frequently in the past week or so.

**Comment by apoelstra** [comment]
rust-miniscript's `parse_descriptor_secret` fuzztest fails because of this

**Comment by tcharding** [comment]
Is the link in the PR description to the correct line of code? I can't see how that code effects the pubkey? Is the munging something to do with `test_cleanup_pk`?

**Comment by apoelstra** [comment]
No, the link is outdated. I should not have linked to `master` but rather whatever specific commit `master` was at hwen I opened this issue..

**Comment by jhfrontz** [comment]
@tcharding asks:
> Is the munging something to do with `test_cleanup_pk`?

Yeah, `test_cleanup_pk()` sanitizes the keys resulting from a deserialization (by "homogenizing" the extra bits in the final byte), making any subsequent re-serialization result in a different blob than was originally deserialized.

I'm guessing that the alternative would be to have the deserialization fail, which would result in much fewer successful parses (and thus less frequent fuzzer coverage of any enclosing data structure's remaining fields).
